### DISCUSSION TITLE
1.39 - WSLg II: WSLg Better

### DISCUSSION THREAD

**Discussion by cerebrate** [arkane-systems/genie#153] [General]
* Handle the WSLg X11 socket and runtime directory better, following Daniel Llewellyn's example here: https://github.com/diddledan/one-script-wsl2-systemd

**NOTE:** The Fedora and Arch installers require manually enabling the _wslg-xwayland.socket_ systemd unit on first launch.

<hr /><em>This discussion was created from the release <a href='https://github.com/arkane-systems/genie/releases/tag/v1.39'>1.39 - WSLg II: WSLg Better</a>.</em>

**Comment by PavelSosin-320** [comment]
In which folder genie is installed now in Fedora? It disappears completely after "successful" installation:
which genie
/usr/bin/which: no genie in (/root/.local/bin:/root/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/root/.dotnet/tools:/usr/libexec/genie)
----------
Even the genie rpm can be downloaded to the local Windows(!) folder and after removal of the previous version using rpm -e. The new version installation using rpm -i /mnt/c/...... Windows DownloadFolder\.... genie.rpm doesn't solve all problems.
env file doesn't take into account that wsl -d aDistro -u root genie -s sets pwd to the Windir, i.e. /mnt/c... . wsl -u doesn't run pam and bash. It is no-login bash:
cat /run/genie.path
/mnt/c/Windows/System32[root@MSI-wsl ~]

**Reply by cerebrate** [reply]
Nothing's been changed in this respect; it should be right where it was. Check out the .spec file.

**Comment by PavelSosin-320** [comment]
Every old genie version installation created new files. Maybe because I used other than rpm method. I had to remove them using rpm. **Starting from 1.40 ** **everything is OK.**
